File: messenger-ui/src/service/ChatService.js
import {$authApi} from "../http";

export default class ChatService {

    static async getChats() {
        return $authApi.get("/chats")
    }

    static async createMessage(chatId, content) {
        return $authApi.post(`/chats/${chatId}/messages`, {content});
    }

    static async editMessage(chatId, id, newContent) {
        return $authApi.patch(`/chats/${chatId}/messages/${id}`, {content: newContent});
    }

    static async deleteMessage(chatId, id) {
        return $authApi.delete(`/chats/${chatId}/messages/${id}`);
    }
}

File: messenger-server/src/main/java/com/communication/messengerserver/service/ChatService.java
package com.communication.messengerserver.service;

import com.communication.messengerserver.entity.Chat;
import com.communication.messengerserver.entity.Message;
import com.communication.messengerserver.entity.User;

import java.util.List;

public interface ChatService {

    List<Chat> getUserChats(String username);

    List<String> getMembersExcept(String chatId, String username);

    Message createMessage(String chatId, String senderName, String content);

    Message editMessage(String chatId, Integer messageId, String content);

    boolean deleteMessage(String chatId, Integer messageId);

    // этого здесь не должно быть
    void createChatForNewUser(User user);
}
